packages/ast-spec/src/base/BaseNodeOrToken.ts

```diff
@@ -0,0 +1,21 @@
+// import type { Node } from '../unions/Node';
+import type { Range } from './Range';
+import type { SourceLocation } from './SourceLocation';
+
+export interface BaseNodeOrToken {
+  /**
+   * The source location information of the node.
+   * @see {SourceLocation}
+   */
+  loc: SourceLocation;
+  /**
+   * @see {Range}
+   */
+  range: Range;
+  /**
+   * The parent node of the current node
+   */
+  // parent?: Node;
```

Comment:
tokens don't ever get a `parent` - so this belongs in the `BaseNode` interface
I never did it - but it might be worth adding a comment here saying that this is added by the types in `@typescript-eslint/types` because `eslint` adds the `parent` whilst it is traversing